### PR TITLE
Remove qmltestrunner dependency

### PR DESCRIPTION
**Pull request by LeonMatthesKDAB**
Use QTest directly, this fixes the build on my Fedora machine, which only provides qmltestrunner linked with Qt5.

### CODE REVIEW

File: examples/qml_extension_plugin/tests/CMakeLists.txt [hunk 1]
```diff
@@ -4,23 +4,29 @@
 #
 # SPDX-License-Identifier: MIT OR Apache-2.0
```
Comment (ahayzen-kdab) [inline examples/qml_extension_plugin/tests/CMakeLists.txt:5]: can you try removing `example_qml_extension_plugin_test|example_qml_features_test|example_qml_minimal_myobject_test` and the comment "qmltestrunner fails to import QtQuick module" from `.github/workflows/github-cxx-qt-tests.yml` so we can see if CI likes it?

File: examples/qml_extension_plugin/tests/main.cpp [hunk 2]
```diff
@@ -0,0 +1,2 @@
+#include <QtQuickTest>
```
Comment (ahayzen-kdab) [inline examples/qml_extension_plugin/tests/main.cpp:1]: missing license info

Comment (LeonMatthesKDAB) [reply examples/qml_extension_plugin/tests/main.cpp:1]: Yup...
I really need to add a local configuration file to `cmps`, then I can finally have this automated :roll_eyes:

Comment (LeonMatthesKDAB) [reply examples/qml_extension_plugin/tests/main.cpp:1]: Maybe I can do that in edu time :thinking:

File: examples/qml_extension_plugin/tests/CMakeLists.txt (hunk 1 above)
Comment (LeonMatthesKDAB) [reply examples/qml_extension_plugin/tests/CMakeLists.txt:5]: Am currently doing that in a separate commit.
Would like to get this working on Linux CI first, then see if it fixes #110

File: examples/qml_extension_plugin/tests/main.cpp [hunk 3]
```diff
@@ -0,0 +1,8 @@
+// SPDX-FileCopyrightText: 2023 Klarälvdalens Datakonsult AB, a KDAB Group
+// company <info@kdab.com> SPDX-FileContributor: Leon Matthes
```
Comment (ahayzen-kdab) [inline examples/qml_extension_plugin/tests/main.cpp:2]: you need the clang format on/off otherwise it corrupts things, see other cpp files :-)

Comment (LeonMatthesKDAB) [reply examples/qml_extension_plugin/tests/main.cpp]: Done

File: examples/qml_extension_plugin/tests/CMakeLists.txt [hunk 4]
```diff
@@ -4,23 +4,38 @@
...
+endif()
+if(NOT Qt6_FOUND)
+    find_package(Qt5 5.15 COMPONENTS QuickTest REQUIRED)
 endif()
+get_target_property(QMAKE Qt::qmake IMPORTED_LOCATION)
 
-set(test_cmd ${QMLTESTRUNNER_BIN} -import "${CMAKE_CURRENT_BINARY_DIR}/../qml" -input ${CMAKE_CURRENT_SOURCE_DIR})
-add_test(NAME ${APP_NAME}_test
+add_executable(${APP_NAME} main.cpp)
+target_link_libraries(${APP_NAME} Qt::QuickTest)
```
[the remark this answers is not in the document]
Comment (LeonMatthesKDAB) [reply examples/qml_extension_plugin/tests/CMakeLists.txt]: Trying that, as well as adding Qt::QuickControls2 and friends to target_link_libraries.

[the remark this answers is not in the document]
Comment (LeonMatthesKDAB) [reply examples/qml_extension_plugin/tests/CMakeLists.txt:18]: That makes it at least find QQuickControls, but it complains that a statically built Qt can't load plugins.
So it seems that this is the actual root cause on macOS.

This PR still fixes the tests on my local machine, so would still be worth keeping IMHO

[the remark this answers is not in the document]
Comment (ahayzen-kdab) [reply examples/qml_extension_plugin/tests/CMakeLists.txt:18]: Right, lets abort the macOS changes

**Review by ahayzen-kdab — APPROVED** [review]
LGTM :-)

**Pull request merged**